operator/pkg/clusterconfiguration/expander.go

```diff
@@ -148,7 +150,18 @@ func ExpandForConfiguration(
 			}
 			properties[k] = value
 		case v.ExternalSecretRef != nil:
-			return nil, fmt.Errorf("unimplemented: ExternalSecretRef for configuration entry %q", k)
+			if cloudExpander == nil {
+				return nil, fmt.Errorf("configuration entry %q: external secret references are unsupported", k)
```

Comment:
nit: I don't think this error message is correct. If cloudExpand == nil it means that external secret references haven't been enabled or they've been disabled depending on how you look it it. This error message makes it seem like something hasn't be connected properly.